### PR TITLE
Trim whitespace of existing assignment and grouping titles

### PR DESCRIPTION
**Pull request by marcospri**
Since https://github.com/hypothesis/lms/pull/6472 we do the same in the python code, we trim new assignment and course titles. We also normalize empty (``) titles to null.

This PR does the same for existing titles in the DB with a DB migration.

### Testing

```
tox -e dev --run-command 'alembic upgrade head'
dev run-test-pre: PYTHONHASHSEED='378749627'
dev run-test-pre: commands[0] | pip-sync-faster requirements/dev.txt --pip-args --disable-pip-version-check
dev run-test: commands[0] | alembic upgrade head
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade f8ff7e49cbbf -> e5a9845d55d7, Clean grouping and assignment titles.
Assignment titles updated: 5
Empty assignment titles: 1
Grouping titles updated: 2
```

### CODE REVIEW

**Comment by marcospri** [comment]
For https://github.com/hypothesis/lms/issues/6456

**Comment by acelaya** [comment]
Could you add a description to the PR?

**Comment by marcospri** [comment]
> Could you add a description to the PR?

My bad, had to many PRs and marked the wrong one ready for review. Added a bit more context now.

**Comment by robertknight** [comment]
When testing this I noticed it is possible to create groups with all-whitespace names (eg. 5 spaces), whereas the old create group form will prevent you from doing this.

File: lms/migrations/versions/e5a9845d55d7_clean_grouping_and_assignment_titles.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def upgrade() -> None:
+    conn = op.get_bind()
+
+    result = conn.execute(
+        sa.text(
+            """
+        UPDATE assignment SET title = TRIM(title)
+        WHERE TRIM(title) != title;
```
Comment (acelaya) [inline lms/migrations/versions/e5a9845d55d7_clean_grouping_and_assignment_titles.py:21]: Does this really make a difference? Couldn't we just get away with `UPDATE assignment SET title = TRIM(title)`?

**Review by acelaya — APPROVED** [review]

File: lms/migrations/versions/e5a9845d55d7_clean_grouping_and_assignment_titles.py (hunk 1 above)
Comment (marcospri) [reply lms/migrations/versions/e5a9845d55d7_clean_grouping_and_assignment_titles.py:21]: Yes, while the where makes the query slower it should lock less rows which should be safer

Comment (acelaya) [reply lms/migrations/versions/e5a9845d55d7_clean_grouping_and_assignment_titles.py:21]: Ah, good point.

**Pull request merged**